lib/browser/api/dialog.js

```diff
@@ -70,147 +70,140 @@ const checkAppInitialized = function () {
         dialogProperties |= fileDialogProperties[prop]
       }
     }
+  }
 
-    if (title == null) {
-      title = ''
-    } else if (typeof title !== 'string') {
-      throw new TypeError('Title must be a string')
-    }
+  if (filters == null) filters = []
 
-    if (buttonLabel == null) {
-      buttonLabel = ''
-    } else if (typeof buttonLabel !== 'string') {
-      throw new TypeError('Button label must be a string')
-    }
+  if (title == null) {
+    title = ''
+  } else if (typeof title !== 'string') {
+    throw new TypeError('Title must be a string')
+  }
 
-    if (defaultPath == null) {
-      defaultPath = ''
-    } else if (typeof defaultPath !== 'string') {
-      throw new TypeError('Default path must be a string')
-    }
+  if (buttonLabel == null) {
+    buttonLabel = ''
+  } else if (typeof buttonLabel !== 'string') {
+    throw new TypeError('Button label must be a string')
+  }
+
+  if (defaultPath == null) {
+    defaultPath = ''
+  } else if (typeof defaultPath !== 'string') {
+    throw new TypeError('Default path must be a string')
+  }
+
+  if (message == null) {
+    message = ''
+  } else if (typeof message !== 'string') {
+    throw new TypeError('Message must be a string')
+  }
 
-    if (filters == null) {
-      filters = []
+  // set up extra parameters for showSaveDialog
+  if (type === 'save') {
+    if (nameFieldLabel == null) {
+      nameFieldLabel = ''
+    } else if (typeof nameFieldLabel !== 'string') {
+      throw new TypeError('Name field label must be a string')
     }
 
-    if (message == null) {
-      message = ''
-    } else if (typeof message !== 'string') {
-      throw new TypeError('Message must be a string')
+    if (showsTagField == null) {
+      showsTagField = true
     }
+  }
 
-    const wrappedCallback = typeof callback === 'function' ? function (success, result, bookmarkData) {
-      return success ? callback(result, bookmarkData) : callback()
-    } : null
-    const settings = {title, buttonLabel, defaultPath, filters, message, securityScopedBookmarks, window}
-    settings.properties = dialogProperties
-    return binding.showOpenDialog(settings, wrappedCallback)
-  },
+  let wrappedCallback = null
+  if (typeof callback === 'function') {
+    wrappedCallback = (success, result, bookmarkData) => success ? callback(result, bookmarkData) : callback()
+  }
 
-  showSaveDialog: function (...args) {
-    checkAppInitialized()
+  let settings
+  if (type === 'open') {
+    settings = {window, buttonLabel, defaultPath, filters, properties, title, message, securityScopedBookmarks, callback}
+  } else {
+    settings = {window, buttonLabel, defaultPath, filters, properties, title, message, securityScopedBookmarks, callback, nameFieldLabel, showsTagField}
+  }
 
-    let [window, options, callback] = parseArgs(...args)
+  return {settings, dialogProperties, wrappedCallback}
+}
+
+module.exports = {
+  showOpenDialog: function (args) {
+    let options = args
```

Comment:
What for? Just rename the `args`  argument if you don't like it )

Reply:
typically in JS it's discouraged to reassign parameters, as that can lead to side effects

Reply:
Probably I don't understand what you mean.

Assigning a new value to a parameter is perfectly fine, it is exactly the same thing as changing value of a regular variable. Changing properties of objects you got in the arguments might not be the best idea, that's true. But using `let options = args` cannot prevent side effects, cause this assignment simply makes both `args` and `options` to "point" to the same object.